main.go

```diff
@@ -67,10 +67,13 @@ func main() {
 	cfg := parseFlags()
 
 	registry := prometheus.NewRegistry()
+	roundTripperInst := newRoundTripperInstrumenter(registry)
 
 	ctx, cancel := context.WithCancel(context.Background())
-	client := &http.Client{}
 	t := http.DefaultTransport.(*http.Transport).Clone()
+	client := &http.Client{
+		Transport: roundTripperInst.NewRoundTripper("thanos-rule-syncer-http", t),
```

Comment:
nit: maybe this name is not the most clear, since OAuth is also over HTTP. For me, labeling the metrics with `{client="thanos-rule-syncer-http"}` is a bit confusing and implies that the other client isn't part of the Thanos Rule Syncer, or something. This client is used for two things:
1. fetching rules; and
2. instructing Thanos to reload its configuration.
Perhaps we could rename the label to:
```suggestion
		Transport: roundTripperInst.NewRoundTripper("fetch-and-reload", t),
```

Or even better, we create a _new_ HTTP client that is used just for Thanos reloading. This client doesn't need to have OAuth on it, and then we can have two different `RoundTripper`s on the clients. One with `{client="fetch"}` and one with `{client="reload"}`.

WDYT?

Reply:
makes sense - I think is a good idea to separate the clients! I've added this change, could you take another look?